pkg/controllers/resourcepack_reconciler.go

```diff
@@ -78,19 +80,18 @@ func WithLongWait(d time.Duration) ResourcePackReconcilerOption {
 
 func NewResourcePackReconciler(m manager.Manager, of schema.GroupVersionKind, options ...ResourcePackReconcilerOption) *ResourcePackReconciler {
 	nr := func() resource.ParentResource {
-		return runtimeresource.MustCreateObject(schema.GroupVersionKind(of), m.GetScheme()).(resource.ParentResource)
+		u := &unstructured.Unstructured{}
+		u.SetGroupVersionKind(of)
+		return u
 	}
-	// Early panic if the resource doesn't satisfy ParentResource interface.
-	_ = nr()
 
 	r := &ResourcePackReconciler{
 		kube:              m.GetClient(),
 		newParentResource: nr,
 		shortWait:         defaultShortWait,
 		longWait:          defaultLongWait,
-		templatingEngine:  kustomize.NewKustomizeEngine(),
+		templatingEngine:  kustomize.NewKustomizeEngine(&kustomizeapi.Kustomization{}),
```

Comment:
Is the idea that we'll use the author-specified Kustomization later?

Reply:
That's for the case where no engine is specified as option when this function is called, which is usually not the case. But if it ever happens, we'll use an empty `Kustomization` object.